packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -2085,6 +2092,36 @@ class _OverlayPortalElement extends RenderObjectElement {
     }
   }
 
+  @override
+  void debugVisitOnstageChildrenWhenOffstage(ElementVisitor visitor, { Element? firstOffstageElement }) {
+    final Element? child = _child;
+    final Element? overlayChild = _overlayChild;
+    if (child != null) {
+      child.debugVisitOnstageChildrenWhenOffstage(visitor, firstOffstageElement: firstOffstageElement ?? this);
+    }
+    if (overlayChild != null) {
+      final _OverlayEntryLocation location = overlayChild.slot! as _OverlayEntryLocation;
+      final _Theater? theater = location._childModel.context.findAncestorWidgetOfExactType<_Theater>();
+      bool overlayIsOnstage = theater != null && theater.skipCount < theater.children.indexOf(location._childModel.widget);
+      if (overlayIsOnstage) {
+        if (firstOffstageElement != null) {
+          visitAncestorElements((Element element) {
+            if (element == location._childModel.context){
```

Comment:
Is there a way to avoid walking back up the tree? If `location._childModel.context` and `firstOffstageElement` are ancestors of this Element, would it be faster if you just compare the depth of the two elements?